lib/luacrypto_aead.c

```diff
@@ -0,0 +1,325 @@
+/*
+* SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+/***
+* Low-level Lua interface to the Linux Kernel Crypto API for AEAD
+* (Authenticated Encryption with Associated Data) ciphers.
+*
+* This module provides a `new` function to create AEAD transform objects,
+* which can then be used for encryption and decryption.
+*
+* @module crypto_aead
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/crypto.h>
+#include <crypto/aead.h>
+#include <linux/err.h>
+#include <linux/scatterlist.h>
+#include <linux/slab.h>
+
+#include <lua.h>
+#include <lualib.h>
+#include <lauxlib.h>
+#include <lunatik.h>
+
+LUNATIK_PRIVATECHECKER(luacrypto_aead_check, struct crypto_aead *);
+
+static void luacrypto_aead_release(void *private)
+{
+	struct crypto_aead *tfm = (struct crypto_aead *)private;
+	if (tfm)
+		crypto_free_aead(tfm);
+}
+
+/***
+* AEAD object methods.
+* These methods are available on AEAD objects created by `crypto_aead.new()`.
+* @see crypto_aead.new
+* @type crypto_aead
+*/
+
+/***
+* Sets the encryption key for the AEAD transform.
+* @function crypto_aead:setkey
+* @tparam string key The encryption key.
+* @raise Error if setting the key fails (e.g., invalid key length for the algorithm).
+*/
+static int luacrypto_aead_setkey(lua_State *L) {
+	struct crypto_aead *tfm = luacrypto_aead_check(L, 1);
+	size_t keylen;
+	const char *key = luaL_checklstring(L, 2, &keylen);
+	int ret = crypto_aead_setkey(tfm, key, keylen);
+	if (ret < 0) {
+		luaL_error(L, "Failed to set AEAD key (err %d)", ret);
+	}
```

Comment:
```suggestion
	if (ret < 0)
		luaL_error(L, "Failed to set AEAD key (err %d)", ret);
```

please mind this style across this patch